clients/java/src/test/java/io/camunda/zeebe/client/ZeebeClientTest.java

```diff
@@ -728,6 +738,125 @@ public void shouldNotSetDefaultJobWorkerTenantIdsFromSetterWithCloudClientBuilde
         .isEqualTo(builder);
   }
 
+  @Test
+  @DisplayName("should throw when gatewayTarget is invalid")
+  public void shouldThrowWhenGatewayTargetIsInvalid() {
+    // given
+    final ZeebeClientBuilderImpl builder = new ZeebeClientBuilderImpl();
+
+    // when
+    Assertions.assertThrows(RuntimeException.class, () -> builder.gatewayTarget("0.0.0.0:26500"));
+  }
+
+  @Test
+  @DisplayName("should use gatewayAddress when gatewayAddress is valid ip:port")
+  public void shouldUseGatewayAddressWhenGatewayAddressIsValidIpPort() {
+    // given
+    final ZeebeClientBuilderImpl builder = new ZeebeClientBuilderImpl();
+    final Properties properties = new Properties();
+    properties.putIfAbsent(ClientProperties.GATEWAY_ADDRESS, "0.0.0.0:26500");
+    builder.withProperties(properties);
+
+    // when
+    builder.build();
+
+    // then
+    Assertions.assertNull(builder.getGatewayTarget());
+
+    Assertions.assertEquals("0.0.0.0:26500", builder.getGatewayAddress());
+  }
```

Comment:
:x: We don't need this test case. The `gatewayAddress` value is the default one, and we already test that in this test class.